MobiusCore/Source/MobiusController.swift

```diff
@@ -234,13 +268,16 @@ public final class MobiusController<Model, Event, Effect> {
             }
         }
 
+        /// Call `closure` with the current state. It will execute on the loop queue.
         func syncRead<T>(_ closure: (Snapshot) throws -> T) rethrows -> T {
+            dispatchPrecondition(condition: .notOnQueue(loopQueue))
             return try loopQueue.sync {
                 try closure(snapshot())
             }
         }
 
-        func syncMutateStopped(error: String, _ closure: (inout StoppedState) -> Void) {
+        /// Mutate the stopped state, assuming we’re currently stopped. If not, fail with the provided error message.
+        func mutateIfStopped(error: String, _ closure: (inout StoppedState) -> Void) {
```

Comment:
Would it be a good idea to rename `error` into `orError`? That would clarify the meaning of the parameter to me, but I'm not sure if it is unidiomatic. Or if 'unidiomatic' is a word for that matter. :)

Reply:
Either way works for me